wireapp/wire-android
# fix: disable swiping in some extra cases [WPB-9044]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9044" title="WPB-9044" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-9044</a>  [Android] Fix multiple replying issues
  </td></table>
  <br />
 






----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

1. Messages in search results and gallery are swipable to reply. And they definitely shouldn't.
2. Deleted messages are a swipable to reply.
3. Self-deleting messages show a slight indication that swiping is possible.

### Causes

![SadSorryGIF](https://github.com/wireapp/wire-android/assets/9389043/197f83bd-2f34-4a29-a9d4-a6258b6fe2e2)

Also, Compose's `SwipeToDismissBox` shows a small indication that the item is swipable even when it is disabled.

### Solutions

Replace the `onSwipeToReply` lambda with a `SwipableMessageConfiguration` sealed interface, that can be either enabled or disabled.

And completely avoid drawing a `SwipeToDismissBox` when the message shouldn't be swipable.

### Testing

Manually tested

#### How to Test

1. Go to conv details, search for stuff and try to swipe.
2. Go to conv details, see the gallery and try to swipe.

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
